pkg/collector/corechecks/containers/generic/filters.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package generic
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// ContainerFilter defines an interface to exclude containers based on Metadata
+type ContainerFilter interface {
+	IsExcluded(container *workloadmeta.Container) bool
+}
+
+// FuncContainerFilter allows any function to be used as a ContainerFilter
+type FuncContainerFilter func(container *workloadmeta.Container) bool
+
+// IsExcluded returns if a container should be excluded or not
+func (f FuncContainerFilter) IsExcluded(container *workloadmeta.Container) bool {
+	return f(container)
+}
+
+// ANDContainerFilter implements a logical AND between given filters
+type ANDContainerFilter struct {
+	Filters []ContainerFilter
+}
+
+// IsExcluded returns if a container should be excluded or not
+func (f ANDContainerFilter) IsExcluded(container *workloadmeta.Container) bool {
+	for _, filter := range f.Filters {
+		if filter.IsExcluded(container) {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
should probably be a sub package in workloadmeta?

Reply:
We need to see what we want to do to with Metadata filtering, but it's out the of the scope of this PR. I'd leave it as-is and change whenever we have something to offer in `workloadmeta`.